Octokit/Models/Response/GitHubApp.cs

```diff
@@ -31,57 +32,62 @@ public GitHubApp(long id, string slug, string name, User owner, string descripti
         /// <summary>
         /// The Id of the GitHub App.
         /// </summary>
-        public long Id { get; private set; }
+        public long Id { get; protected set; }
```

Comment:
What's the reason for (and the ramifications of) the permissions change here?

Reply:
To support inheritance.

`GitHubAppFromManifest` inherits from `GitHubApp`.  The `SimpleJson`/`SimpleJsonSerializer` this project uses won't populate any of the  properties inherited from `GitHubApp` on `GitHubAppFromManifest` objects if the property is `private set;` (i.e. `id`, `slug`, `name`, etc. are all `null`).

Another example is [`License`](https://github.com/octokit/octokit.net/blob/main/Octokit/Models/Response/License.cs) subclasses [`LicenseMetadata`](https://github.com/octokit/octokit.net/blob/main/Octokit/Models/Response/LicenseMetadata.cs).  All properties on `LicensesMetadata` are `protected set`

As far as ramifications, I can't think of any that would be negative.  From the docs:

> Only code in the same class or in a derived class can access this type or member.

Honestly, it would be a big help if the standard in this library was to use `protected set` instead of `private set`.  I've often tried to create subclasses of types from this library ( like`Repository` or `GitHubApp`) to quickly add and deserialize properties being returned by the API but missing from this library's types.  Unfortunately, this is not possible due to the issue above.

Reply:
This is probably one of the best arguments for using `protected` that I have heard so far.  Just for some context on why they are all private, you can read this [issue](https://github.com/octokit/octokit.net/issues/2472) and see this [PR](https://github.com/octokit/octokit.net/pull/2565) and [release](https://github.com/octokit/octokit.net/discussions/2567) if you're interested. 

The gist was that the library was pretty inconsistent when it came to how things were implemented and even a couple of years ago we knew we wanted to head to a generative approach.  We went with the more restrictive pattern of using `private` accessors as a clean slate starting point so that we'd have some consistently and so that we could generate source using that consistency.  

Given that we are close to GA'ing our [Generated .NET SDK](https://github.com/octokit/dotnet-sdk) and for the points you made, it makes sense to make  mutating these properties more permissive.

Please feel free to push forward on that front if you have time.  As a word of caution, the custom serializer that has been built makes use of property accessors and you might run into a couple issues around the serialization tests - but they should be relatively low risk.